### PR TITLE
Update testing

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose

This PR contains two main changes:
1. renamed `test` directory to `tests`, which is more consistent with our other repos
2. no longer install in-place and test in the root directory

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Checklist

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by marcomangano — APPROVED** [review]

**Review by ArshSaja — DISMISSED** [review]

**Comment by sseraj** [comment]
The changes look straightforward, but why does the coverage drop 30%?

**Comment by nwu63** [comment]
You know I'm not totally sure. My guess is that the change in file path (due to `cd tests`) meant that the `.coveragerc` file somehow stopped working. I'll try to see if I can fix it.

**Comment by nwu63** [comment]
@sseraj the coverage has been fixed.

**Review by sseraj — APPROVED** [review]

File: .github/test_real.sh [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 set -e
 
-# all tests should pass on private
+# all tests should pass on the private image
 if [[ $IMAGE == "private" ]]; then
     EXTRA_FLAGS='--disallow_skipped'
 fi
 
+cd tests
+# we have to copy over the coveragerc file to make sure it's
```
Comment (marcomangano) [inline .github/test_real.sh:10]: it's what? :)

**Review by marcomangano — APPROVED** [review]

**Pull request merged**